DRILL-7456: Batch count fixes for 12 operators

paul-rogers opened the pull request:
Enables batch validation for 12 additional operators:

* MergingRecordBatch
* OrderedPartitionRecordBatch
* RangePartitionRecordBatch
* TraceRecordBatch
* UnionAllRecordBatch
* UnorderedReceiverBatch
* UnpivotMapsRecordBatch
* WindowFrameRecordBatch
* TopNBatch
* HashJoinBatch
* ExternalSortBatch
* WriterRecordBatch

Fixes issues found with those checks so that this set of
operators passes all checks.

Includes code cleanup in many files touched during this
work.

arina-ielchiieva commented on exec/java-exec/src/test/java/org/apache/drill/exec/DrillSeparatePlanningTest.java:
```diff
@@ -127,6 +127,7 @@ public void testPlanningNoSplit() throws Exception {
       PlanFragment rootFragment = planFragments.getFragments(0);
       assertFalse(rootFragment.getLeafFragment());
 
+      client.queryBuilder().plan(planFragments.getFragmentsList()).print();
```
Not sure we should print here... This would poppet test log.

arina-ielchiieva commented on exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/agg/TestHashAggrSpill.java:
```diff
@@ -126,7 +126,8 @@ private void runAndDump(ClientFixture client, String sql, long expectedRows, lon
+   * Test Secondary and Tertiary spill cycles - Happens when some of the spilled
+   * partitions cause more spilling as they are read back
    *
    * @throws Exception
```
Please remove `@throws` or describe the exception.

arina-ielchiieva commented on exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/flatten/TestFlatten.java:
```diff
@@ -397,7 +396,6 @@ public void testDrill1665() throws Exception {
     if(RUN_ADVANCED_TESTS){
```
space -> `if (RUN_ADVANCED_TESTS) {`

paul-rogers commented:
@arina-ielchiieva, thanks much for the review. Addressed comments. Reverted (most) changes to header files. Have disabled automatic import organization in my IDE; will deal with maintaining imports manually until we decide on a standard order.

arina-ielchiieva commented:
+1, LGTM
